### ISSUE TITLE
Filenames in utf-8 cannot be opened or written after upgrade to 1.2.3

### ISSUE THREAD

**Issue by ethomson** [Azure/azure-storage-fuse#417]
### Which version of the blobfuse was used?

blobfuse 1.2.3

### Which OS (please include version) are you using?

Ubuntu 18.04 (Linux hermes 5.3.0-1022-azure #23~18.04.1-Ubuntu SMP Mon May 11 11:55:56 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux)

### What problem was encountered?

I upgraded to blobfuse 1.2.3, now my volume no longer supports filenames in UTF-8.

Trying to create a file with a UTF-8 filename gives "Permission denied".

```
# touch Æ
touch: cannot touch 'Æ': Permission denied
```

On a volume with filenames already created:

```
# ls -Flas
ls: cannot access 'é': Permission denied
total 0K
 ? -????????? ? ?    ?             ?            ?  é
# cat <0301>
cat: é: Permission denied
```

This is not a permission problem; other files on the volume work as normal.

### Have you found a mitigation/solution?

Downgrade to an earlier version.  blobfuse 1.0.2 supports this correctly.

### By default, blobfuse logs errors to syslog. If this is relevant, is there anything in the syslog that might be helpful?

```
Jun  6 16:36:00 hermes blobfuse[26878]: Sat Jun  6 16:36:00 2020 ==> REQUEST/RESPONSE#012#011GET https://MYVOLUME.blob.core.windows.net/media?comp=list&delimiter=/&include=metadata&maxresults=2&prefix=file/é&restype=container#012#01
1User-Agent: [Azure-Storage-Fuse/1.2.3]#012#011x-ms-date: [Sat, 06 Jun 2020 16:36:00 GMT]#012#011x-ms-version: [2018-11-09
]#012#011Authorization: [REDACTED]#012#011Transfer-Encoding: []#012#011---------------------------------------------------
-----------------------------#012#011RESPONSE Status: 403#012#011Content-Length: [744]#012#011Content-Type: [application/x
ml]#012#011Date: [Sat, 06 Jun 2020 16:35:59 GMT]#012#011Server: [Microsoft-HTTPAPI/2.0]#012#011x-ms-error-code: [Authentic
ationFailed]#012#011x-ms-request-id: [082cef6c-801e-0125-4f20-3c0600000000]
Jun  6 16:36:00 hermes blobfuse[26878]: Failure when attempting to determine if file/é exists on the service.
errno = 403.
```

### If relevant, please share your mount command.

```
blobfuse /data --tmp-path=/mnt/blobtmp --config-file=/creds -o attr_timeout=240 -o entry_timeout=240 -o negative_timeout=120 -o allow_other --log-level=LOG_DEBUG
```

**Comment by ethomson** [comment]
Update: 1.1.1 is the last version that supports this correctly.

**Comment by vibhansa-msft** [comment]
This issue has been root-caused. Unicode character encoding is having issue and as a result URI constructed to access the blob is incorrect. Working on the fix.

**Comment by vibhansa-msft** [comment]
fixed in 1.3.1

**Comment by ethomson** [comment]
Thanks!
